pkgs/applications/science/misc/boinc/default.nix

```
@@ -40,24 +41,25 @@ stdenv.mkDerivation rec {
 
   buildInputs = [
     curl
+    sqlite
+    patchelf
+  ] ++ lib.optionals (!headless) [
     libGLU
     libGL
     libXmu
     libXi
     freeglut
     libjpeg
     wxGTK32
-    sqlite
     gtk3
     libXScrnSaver
     libnotify
-    patchelf
     libX11
     libxcb
     xcbutil
   ];
 
-  NIX_LDFLAGS = "-lX11";
+  NIX_LDFLAGS = lib.optionals (!headless) "-lX11";
```

```suggestion
  NIX_LDFLAGS = lib.optionalString (!headless) "-lX11";
```
Is there any reason why you used `lib.optionals` instead of `lib.optionalString` or was this a mistake? I think `NIX_LDFLAGS` should be a string and not an array.

---

This was a mistake, you're right. `NIX_LDFLAGS` should be a string